astroquery/noirlab/core.py

```diff
@@ -90,12 +102,138 @@ def query_region(self, coordinate, radius=0.1, cache=True):
+    # #!        string or as the appropriate `~astropy.coordinates` object.
+    # #!    radius : str or `~astropy.units.Quantity` object, optional
+    # #!        Default 0.1 degrees.
+    # #!        The string must be parsable by `~astropy.coordinates.Angle`.The
+    # #!        appropriate `~astropy.units.Quantity` object from
+    # #!        `~astropy.units` may also be used.
+    # #!
+    # #!    Returns
+    # #!    -------
+    # #!    response : `~astropy.table.Table`
+    # #!    """
+    # #!    self._validate_version()
+    # #!    ra, dec = coordinate.to_string('decimal').split()
+    # #!    url = f'{self.siaurl}?POS={ra},{dec}&SIZE={radius}&format=json'
+    # #!    response = self._request('GET', url,
+    # #!                             timeout=self.TIMEOUT,
+    # #!                             cache=cache)
+    # #!    response.raise_for_status()
+    # #!    return astropy.table.Table(data=response.json())
+
+    def query_region_async(self, coordinate, radius=0.1, cache=True):
+        """Query for NOIRLab observations by region of the sky.
+
+        Given a sky coordinate and radius, returns a `~astropy.table.Table`
+        of NOIRLab observations.
+
+        Parameters
+        ----------
+        coordinates : str or `~astropy.coordinates` object
+            The target region which to search. It may be specified as a
+            string or as the appropriate `~astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.1 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used.
+
+        Returns
+        -------
+        response : `requests.Response`
+        """
+        self._validate_version()
+
+        ra, dec = coordinate.to_string('decimal').split()
+        url = f'{self.siaurl}?POS={ra},{dec}&SIZE={radius}&format=json'
+        response = self._request('GET', url,
+                                 timeout=self.TIMEOUT,
+                                 cache=cache)
+        response.raise_for_status()
+        return response
+
+    def core_fields(self, cache=True):
+        """List the available CORE fields. CORE fields are faster to search
+        than AUX fields.."""
+        response = self._request('GET', self._adsc_url,
+                                 timeout=self.TIMEOUT,
+                                 cache=cache)
+        response.raise_for_status()
+        return response.json()
+
+    def aux_fields(self, instrument, proctype, cache=True):
+        """List the available AUX fields. AUX fields are ANY fields in the
+        Archive FITS files that are not core DB fields.  These are generally
+        common to a single Instrument, Proctype combination. AUX fields are
+        slower to search than CORE fields. """
+        url = f'{self._adsa_url}/{instrument}/{proctype}/'
```

Comment:
since `instrument` and `proctype` are user-entered parameters, please document them here (and maybe consider adding some sort of validity check)

Reply:
All of these functions are supported by web-services which do the validation.  We have people accessing directly via the web-services.  Doing validation here would be redundant (and require DB access since what is valid depends on DB.  I've added note in doc that says the acceptable values come from CATEGORICALS method.

Reply:
We need the parameters documented here, in the docstring.

If a function takes parameters, as this one does, those parameters need to be documented in the docstring, i.e.:
```
Parameters
------------
instrument : str
    Instrument name, see <categoricals?> for details
```
etc.